### ISSUE TITLE
GP prior over per-site calibration parameters: wiring into the parameter vector

### ISSUE THREAD

**Issue by arob5** [arob5/spatial-lsm-calibration#33]
Records how a Gaussian process prior over the per-site copies of a calibration parameter wires into the parameter vector redesign (vault: `logs/2026-09-23_Parameter Vector Redesign.md`, sections 5.2, 9 and question 12), so the design decisions taken on 2026-09-23 are not lost before the spatial-prior work starts. The parameter-vector PR (`refactor/parameter-vector`) admits the prior *shape* this needs and proves it with a plain multivariate normal; everything below the line "Not in the parameter-vector PR" is this issue.

## The design decision

A `CalibrationParameter`'s prior is a TFP distribution in natural space whose `.distribution` is the unconstrained base and whose `.bijector` is the transform to natural space. Today every prior is *independent copies*: batch shape `()` or `(n_groups,)`, event shape that of one copy. The redesign also admits a prior that is **joint over groups**: batch shape `()`, event shape `(n_groups,)` or `(n_groups, k)`, one distribution over all the copies at once. A GP over the per-site copies is exactly that, with `varies_by="site"`. The layout does not change, because each parameter already owns one contiguous `(n_groups, size)` block of theta, which is the block a covariance operator wants.

The `ParameterVector` never learns it is holding a GP. Per method:

| method | independent copies | joint over groups |
|---|---|---|
| `log_prior` | `log_prob` per group, summed | `distribution.log_prob(block)`, block `(..., S)` or `(..., S, k)` |
| `sample` | batched draw, `(n, n_groups, size)` | one draw `(n, S)`, reshaped to `(n, S, 1)` |
| `gaussian_prior` | `PSDDiagonal` over groups | `DensePSD` of `distribution.covariance()`, mean `distribution.mean()`; exact |
| `fields`, `flat`, `sipnet_table` | bijector then gather onto sites | unchanged: the bijector is elementwise and a per-site parameter's gather is the identity |
| `select(sites=)` | slice the batch by integer index | the Gaussian marginal: slice mean and covariance, rebuild a `MultivariateNormalTriL` (TFP does not slice event dimensions) |
| `check_map_image_is_in_domain` | corners per group | unchanged, elementwise |
| `describe` | mean/sd per column | `mean()`/`stddev()` per column; natural quantiles blank, as for any vector event |

## Where the GP is built

In a prior helper beside `log_normal` and the others, not in `CalibrationParameter`:

```python
def gp_log_normal(site_table, *, kernel, mean_fn):
    x = jnp.asarray(SITE_PROJECTION.forward(site_table["lon"], site_table["lat"])).T  # (S, 2) meters
    gp = tfd.GaussianProcess(kernel, index_points=x, mean_fn=mean_fn)                  # MVN over S sites, unconstrained
    return tfd.TransformedDistribution(gp, tfb.Exp())                                   # event (S,), natural space
```

- **No GP package is needed for the prior itself.** `tfd.GaussianProcess(kernel, index_points, mean_fn)` with a kernel from `tfp.math.psd_kernels` (`MaternThreeHalves`, `ExponentiatedQuadratic`, ...) runs on the JAX substrate and is a multivariate normal over the index points with analytic `.mean()` and `.covariance()`. Verified 2026-09-23 on `tfp-nightly 0.26.0.dev20260922`: wrapped in `TransformedDistribution(..., Exp())` it has event shape `(S,)`, batch shape `()`, and a positive natural-space image. Built without `jax_enable_x64` it comes out `float32`, which `check_prior_is_float64` refuses; the helper must build in float64.
- **Spatial coordinates.** The helper projects lon/lat through `sipnet_calibration.projection.SITE_PROJECTION` (Lambert azimuthal equal-area, 50 N 100 W) to planar meters. The projection's anisotropy stays under 1.3 over the whole pool (asserted in `tests/test_projection.py`), so a stationary kernel on the projected plane is honest. A kernel on the sphere (great-circle or chordal distance) is a later swap inside the helper and changes nothing outside it.
- **Covariates in the mean.** Arguments of the helper, not of the vector: `mean_fn = lambda pts: X @ beta`, with `X` the covariate matrix aligned to the site table's rows (the tracked `data/raw/covariates/` product joined on `site_id`; a site-labels class one-hot encoded when used as a categorical covariate) and `beta` fixed. The `CalibrationParameter` sees a distribution and nothing else.
- **Scale.** If dense solves at S = 8000 become the bottleneck (a `(8000, 8000)` covariance is 512 MB in float64, and `gaussian_prior` needs its factor once per run), `tinygp` or a pyEKI structured `PSDLinOp` (Matern operator, low-rank factor) can produce the same multivariate normal. What the `CalibrationParameter` stores is still a TFP distribution; the seam for reading a structured operator off the prior instead of calling `.covariance()` is `ParameterVector._moment_block`.
- **Hyperparameters.** An unknown `beta`, amplitude or length scale makes the prior conditional on hyperparameters `phi`. That is a separate seam (a `JointDistribution` or a callable prior threaded through `log_prior`/`sample`) and is out of scope here; it is named in section 9 of the design doc.

## Not in the parameter-vector PR

That PR only admits the joint shape (`check_prior_shape_matches_groups`) and implements the right-hand column of the table, tested with a plain `tfd.MultivariateNormalTriL` over the three example sites through `Exp`. This issue is the rest:

- [ ] `gp_log_normal` (and a logit-space sibling if a fraction is ever spatial) in the prior helpers, with `provenance` conventions for a kernel choice
- [ ] the kernel family and its length scale / amplitude, with the elicitation or fit recorded (readiness report section 5.3: theta_s = mu_PFT(s) + u_s + eta_s, u a Matern GP over projected site coordinates)
- [ ] covariates: which columns of `data/raw/covariates/`, one-hot site-labels classes, and the fixed `beta`
- [ ] `select(sites=)` marginalization for a joint prior
- [ ] a structured covariance operator in `gaussian_prior` if the dense block is too slow at the pool scale
- [ ] hyperparameters `phi`, tracked separately once the above exists

Related: #11 (a spatially ordered site column would let a structured operator exploit locality).
